src/Desktop.Analyzers/UnitTests/DoNotUseInsecureDTDProcessingXmlDocumentSetInsecureResolverTests.cs

```diff
@@ -0,0 +1,455 @@
+using System.Xml;
+
+namespace TestNamespace
+{
+    class TestClass
+    {
+        private static void TestMethod(XmlDocument doc)
+        {
+            doc.XmlResolver = null;
+        }
+    }
+}"
+            );
+
+            VerifyBasic(@"
+Imports System.Xml
+
+Namespace TestNamespace
+	Class TestClass
+		Private Shared Sub TestMethod(doc As XmlDocument)
+			doc.XmlResolver = Nothing
+		End Sub
+	End Class
+End Namespace");
+        }
+
+        [Fact]
+        public void XmlDocumentNoCtorUseSecureResolverShouldNotGenerateDiagnostic()
+        {
+            VerifyCSharp(@"
+using System.Xml;
+
+namespace TestNamespace
+{
+    class TestClass
+    {
+        private static void TestMethod(XmlDocument doc, XmlSecureResolver resolver)
+        {
+            doc.XmlResolver = resolver;
+        }
+    }
+}"
+            );
+
+            VerifyBasic(@"
+Imports System.Xml
+
+Namespace TestNamespace
+	Class TestClass
+		Private Shared Sub TestMethod(doc As XmlDocument, resolver As XmlSecureResolver)
+			doc.XmlResolver = resolver
+		End Sub
+	End Class
+End Namespace");
+        }
+        
+        [Fact]
+        public void XmlDocumentNoCtorUseSecureResolverWithPermissionsShouldNotGenerateDiagnostic()
+        {
+            VerifyCSharp(@"
+using System.Net;
+using System.Security;
+using System.Security.Permissions;
+using System.Xml;
+
+namespace TestNamespace
+{
+    class TestClass
+    {
+        private static void TestMethod(XmlDocument doc)
+        {
+            PermissionSet myPermissions = new PermissionSet(PermissionState.None);
+            WebPermission permission = new WebPermission(PermissionState.None);
+            permission.AddPermission(NetworkAccess.Connect, ""http://www.contoso.com/"");
+            permission.AddPermission(NetworkAccess.Connect, ""http://litwareinc.com/data/"");
+            myPermissions.SetPermission(permission);
+            XmlSecureResolver resolver = new XmlSecureResolver(new XmlUrlResolver(), myPermissions);
+
+            doc.XmlResolver = resolver;
+        }
+    }
+}"
+            );
+
+            VerifyBasic(@"
+Imports System.Net
+Imports System.Security
+Imports System.Security.Permissions
+Imports System.Xml
+
+Namespace TestNamespace
+	Class TestClass
+		Private Shared Sub TestMethod(doc As XmlDocument)
+			Dim myPermissions As New PermissionSet(PermissionState.None)
+			Dim permission As New WebPermission(PermissionState.None)
+			permission.AddPermission(NetworkAccess.Connect, ""http://www.contoso.com/"")
+            permission.AddPermission(NetworkAccess.Connect, ""http://litwareinc.com/data/"")
+            myPermissions.SetPermission(permission)
+            Dim resolver As New XmlSecureResolver(New XmlUrlResolver(), myPermissions)
+
+            doc.XmlResolver = resolver
+        End Sub
+    End Class
+End Namespace");
+        }
+
+        [Fact]
+        public void XmlDocumentNoCtorSetResolverToNullInTryClauseShouldNotGenerateDiagnostic()
```

Comment:
I'm a little confused by this sequence of tests. This test says that setting `XmlResolver` to `null` in a `try` block is OK, The next tests say that setting `XmlResolver` to an _insecure resolver_ is bad in a `catch` block or in a _finally_ block.

Is it ok to set `XmlResolver` to `null` in a `catch` block or a `finally` block? Is it bad to set `XmlResolver` to an insecure resolver in a `try` block?

Is there anything special about `try`/`catch`/`finally` with respect to this analyzer? Or were you just seeing what happened when you nested the offending line of code in some other syntactic construct?